cloud/amazon/ec2_vpc_route_table.py

```diff
@@ -319,7 +319,8 @@ def ensure_routes(vpc_conn, route_table, route_specs, propagating_vgw_ids,
     # VGWs in place.
     routes_to_delete = [r for r in routes_to_match
                         if r.gateway_id != 'local'
-                        and r.gateway_id not in propagating_vgw_ids]
+                        and (propagating_vgw_ids is None or
+                             r.gateway_id not in propagating_vgw_ids)]
```

Comment:
Should this be:

```py
propagating_vgw_ids is not None
and r.gateway_id not in propagating_vgw_ids
```
? Right now it looks like this would delete all routes if `propagating_vgw_ids` is `None`. Are we sure that's the intention?